test/fft_utest.cpp

```diff
@@ -140,6 +144,68 @@ class FftTest : public ::testing::Test {
             ASSERT_EQ(ifft_1, v);
         }
     }
+
+    /**
+     * convert a number into a vector of digits padded with zeros
+     *
+     * @param num
+     *
+     * @return
+     */
+    quadiron::vec::Vector<T>*
+    _convert_string2vec(const gf::Field<T>& gf, int n, char num[])
+    {
+        int i;
+        quadiron::vec::Vector<T>* vec = new quadiron::vec::Vector<T>(gf, n);
+        int len = strlen(num);
+        for (i = 0; i < len; i++) {
+            vec->set(i, num[len - i - 1] - '0');
+        }
+        for (; i < n; i++) {
+            vec->set(i, 0);
+        }
+        return vec;
+    }
+
+    /** compute sum of 2 large numbers in string representation
+     *
+     * @param str1 first string
+     * @param str2 second string
+     *
+     * @return the sum string
+     */
+    std::string big_num_add(std::string str1, std::string str2)
+    {
+        if (str1.length() > str2.length())
+            std::swap(str1, str2);
+
+        std::string str = "";
+
+        int n1 = str1.length(), n2 = str2.length();
+
+        std::reverse(str1.begin(), str1.end());
```

Comment:
Instead of reversing the string you  could simply use reverse iterator to walk though the strings (using `rbegin` and `rend`).

Reply:
What about this comment?

It's weird to have a `add` operation that modifies its input, no?

Reply:
pass now by value. this is just a test and a straightforward addition. Don't want to optimize here.

Reply:
Ok, since it's only for test it's fine by me